component/config.jsonnet

```diff
@@ -1,17 +1,31 @@
 local kap = import 'lib/kapitan.libjsonnet';
 local kube = import 'lib/kube.libjsonnet';
+local prometheus = import 'lib/prometheus.libsonnet';
 local inv = kap.inventory();
 local params = inv.parameters.argocd;
 
-local namespace = kube.Namespace(params.namespace) {
-  metadata+: {
-    labels+: {
-      SYNMonitoring: 'main',
-      'app.kubernetes.io/part-of': 'argocd',
-      'openshift.io/cluster-monitoring': 'true',
-    },
-  },
-};
+local namespace =
+  if std.member(inv.applications, 'prometheus') then
+    prometheus.RegisterNamespace(kube.Namespace(params.namespace)) {
```

Comment:
I'd move the object with `metadata: ...` to a variable so it does not need repetition

Reply:
Moved to new variable